lightning/src/chain/onchaintx.rs

```diff
@@ -673,14 +711,33 @@ impl<ChannelSigner: Sign> OnchainTxHandler<ChannelSigner> {
 						// outpoints to know if transaction is the original claim or a bumped one issued
 						// by us.
 						let mut set_equality = true;
-						if request.outpoints().len() != tx.input.len() {
-							set_equality = false;
+						if !request.requires_external_funding() ||
```

Comment:
can't this be simplified to `!request.requires_external_funding() || !request.is_malleable()`?